cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -253,8 +219,47 @@ func (o *CloudControllerManagerOptions) Config(allControllers, disabledByDefault
 		return nil, fmt.Errorf("error creating self-signed certificates: %v", err)
 	}
 
-	c := &cloudcontrollerconfig.Config{}
-	if err := o.ApplyTo(c, CloudControllerManagerUserAgent); err != nil {
+	kubeconfig, err := clientcmd.BuildConfigFromFlags(o.Master, o.Kubeconfig)
```

Comment:
why was this block of code moved. IMO with the split of logic into Config and ApplyTo it's more complex now.

Reply:
revert now